source/index.ts

```diff
@@ -105,60 +112,60 @@ namespace is { // tslint:disable-line:no-namespace
 	const isObject = (value: any) => typeof value === 'object';
 
 	// tslint:disable:variable-name
-	export const undefined = isOfType('undefined');
-	export const string = isOfType('string');
-	export const number = isOfType('number');
-	export const function_ = isOfType('function');
-	export const null_ = (value: any) => value === null;
+	export const undefined = isOfType<undefined>('undefined');
+	export const string = isOfType<string>('string');
+	export const number = isOfType<number>('number');
+	export const function_ = isOfType<Function>('function');
+	export const null_ = (value: any): value is null => value === null;
 	export const class_ = (value: any) => function_(value) && value.toString().startsWith('class ');
-	export const boolean = (value: any) => value === true || value === false;
-	export const symbol = isOfType('symbol');
+	export const boolean = (value: any): value is boolean => value === true || value === false;
+	export const symbol = isOfType<Symbol>('symbol');
 	// tslint:enable:variable-name
 
 	export const array = Array.isArray;
 	export const buffer = Buffer.isBuffer;
 
-	export const nullOrUndefined = (value: any) => null_(value) || undefined(value);
+	export const nullOrUndefined = (value: any): value is null | undefined => null_(value) || undefined(value);
 	export const object = (value: any) => !nullOrUndefined(value) && (function_(value) || isObject(value));
-	export const iterable = (value: any) => !nullOrUndefined(value) && function_(value[Symbol.iterator]);
-	export const generator = (value: any) => iterable(value) && function_(value.next) && function_(value.throw);
+	export const iterable = <T = any>(value: any): value is Iterator<T> => !nullOrUndefined(value) && function_(value[Symbol.iterator]);
```

Comment:
This allows you to write the following

```ts
if(is.iterable<string>(foo)) {
   const result = foo.next();
   // `result` is now an `IteratorResult` of type `string`
}
```

If you want I can remove the generic though as it's not really part of the check. It actually just infers the type of the items inside the Iterator.